### PR TITLE
* #20: Provided Dockerfile for building docker images.

### LINKED ISSUE #20: Provide a docker image
## Problem

We need an easy way to start the API server. We want to provide a Dockerfile for our users.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #20

### CODE REVIEW

File: .github/workflows/linux-build.yml [hunk 1]
```diff
@@ -20,8 +20,11 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Build Docker Image
+        run: docker build --tag rest-api-test-image:latest .
+
       - name: Install Swag
-        run: go get -u github.com/swaggo/swag/cmd/swag
+        run: go install github.com/swaggo/swag/cmd/swag@latest
```
Comment (jakobbraun) [inline .github/workflows/linux-build.yml:27]: Maybe replace `latest` by a specific version?

File: doc/user_guide/user-guide.md [hunk 2]
```diff
@@ -12,49 +12,57 @@ GRANT CREATE SESSION TO api_service_account;
...
-Exasol service account described above:
+Exasol service account described above and a list of API tokens (see [Authorization](#authorization)):
 
 ```yaml
-ExasolUser: "api_service_account"
-ExasolPassword: "secret_password"
+EXASOL_USER: "api_service_account"
+EXASOL_PASSWORD: "secret_password"
+API_TOKENS:
+  - "fwe3cqzE9pQblAYbLFRmxtN03uMgJ2"
```
Comment (jakobbraun) [inline doc/user_guide/user-guide.md:26]: I somehow fear that someone copies it and forgets to change. Maybe add a comment or add -just-an-example- inside the tokens.

File: doc/user_guide/user-guide.md [hunk 3]
```diff
@@ -12,49 +12,57 @@ GRANT CREATE SESSION TO api_service_account;
...
-api-tokens:
-  - "fwe3cqzE9pQblAYbLFRmxtN03uMgJ2"
-  - "nKBwcSyMHr1BnYsV8kiaU0cxNY6iQr"
-  - "ubwl5sCao6RHE3iCqe72M6zJc1cHHQ"
-```
+#### Properties Reading Chain
+
+1. The properties from the YAML file are read.
+2. The configuration from environment variables are read, they override the configurations from the YAML file.
+3. The default values are added if they are not added
```
Comment (jakobbraun) [inline doc/user_guide/user-guide.md:46]:
```suggestion
3. The default values are added if they were not added in a previous step
```

File: doc/user_guide/user-guide.md (hunk 2 above)
Comment (jakobbraun) [reply doc/user_guide/user-guide.md:26]: Or different idea: Add a check in the implementation that blocks these three token and says that they are just examples and the user should use a custom one.

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: .github/workflows/linux-build.yml (hunk 1 above)
Comment (AnastasiiaSergienko) [reply .github/workflows/linux-build.yml:27]: done

File: doc/user_guide/user-guide.md (hunk 2 above)
Comment (AnastasiiaSergienko) [reply doc/user_guide/user-guide.md]: I made them short in the example. If somebody copies it, it just won't work because this length is not allowed

**Review by jakobbraun — APPROVED** [review]

**Review by pj-spoelders — COMMENTED** [review]
Ok for me. Looks good.

**Review by pj-spoelders — APPROVED** [review]

**Pull request merged**